docs/design/federation-phase-1.md

```diff
@@ -157,23 +168,42 @@ clusters over AWS clusters" or "spread load equally across clusters").
 
 ## Ubernetes Cluster Controller
 
-The cluster controller
-performs the following two kinds of work:
-
-1. It watches all the sub-resources that are created by Ubernetes
-   components, like a sub-RC or a sub-service. And then it creates the
-   corresponding API objects on the underlying K8S clusters.
-1. It periodically retrieves the available resources metrics from the
-   underlying K8S cluster, and updates them as object status of the
-   `cluster` API object.  An alternative design might be to run a pod
-   in each underlying cluster that reports metrics for that cluster to
-   the Ubernetes control plane.  Which approach is better remains an
-   open topic of discussion.
+The cluster controller's job is to reconcile the desired state of the
+underlying Kubernetes clusters (as recorded by Ubernetes in it's
+persistent store of desired objects) against the real Kubernetes
+clusters' states.
+
+To achieve this, it performs the following three kinds of work:
+
+1. It watches the Ubernetes persistent store for new or changed
```

Comment:
Do we want to have a single controller that watches all resources? or a controller for every resource? (like we will have ServiceController and ReplicaSetController only for now). Apart from creating or modifying the corresponding k8s resources, these controller will also set appropriate fields in the u8s resources (for ex: ReplicaSetController will set the ObservedGeneration for ReplicaSet, when it observes a replicaset update)